adapters/syslog/syslog.go

```diff
@@ -71,6 +74,80 @@ func getHostname() string {
 	return hostname
 }
 
+func getFieldTemplates(route *router.Route) (*FieldTemplates, error) {
+	var err error
+	var s string
+	var tmpl FieldTemplates
+
+	s = cfg.GetEnvDefault("SYSLOG_PRIORITY", "{{.Priority}}")
+	if tmpl.priority, err = template.New("priority").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting priority to:", s)
+
+	s = cfg.GetEnvDefault("SYSLOG_TIMESTAMP", "{{.Timestamp}}")
+	if tmpl.timestamp, err = template.New("timestamp").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting timestamp to:", s)
+
+	s = getHostname()
+	if tmpl.hostname, err = template.New("hostname").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting hostname to:", s)
+
+	s = cfg.GetEnvDefault("SYSLOG_TAG", "{{.ContainerName}}"+route.Options["append_tag"])
+	if tmpl.tag, err = template.New("tag").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting tag to:", s)
+
+	s = cfg.GetEnvDefault("SYSLOG_PID", "{{.Container.State.Pid}}")
+	if tmpl.pid, err = template.New("pid").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting pid to:", s)
+
+	s = cfg.GetEnvDefault("SYSLOG_STRUCTURED_DATA", "")
+	if route.Options["structured_data"] != "" {
+		s = route.Options["structured_data"]
+	}
+	if s == "" {
+		s = "-"
+	} else {
+		s = fmt.Sprintf("[%s]", s)
+	}
+	if tmpl.structuredData, err = template.New("structuredData").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting structuredData to:", s)
+
+	s = cfg.GetEnvDefault("SYSLOG_DATA", "{{.Data}}")
+	if tmpl.data, err = template.New("data").Parse(s); err != nil {
+		return nil, err
+	}
+	debug("setting data to:", s)
+
+	return &tmpl, nil
+}
+
+func getTCPFraming() (TCPFraming, error) {
+	switch s := cfg.GetEnvDefault("SYSLOG_TCP_FRAMING", string(TraditionalTCPFraming)); s {
+	case string(TraditionalTCPFraming):
+		return TraditionalTCPFraming, nil
+	case string(OctetCountedTCPFraming):
+		return OctetCountedTCPFraming, nil
+	default:
+		return TraditionalTCPFraming, fmt.Errorf("unknown SYSLOG_TCP_FRAMING value: %s", s)
+	}
+}
+
+func getRetryCount() (uint, error) {
+	retryCount, err := strconv.Atoi(cfg.GetEnvDefault("RETRY_COUNT", "10"))
+	return uint(retryCount), err
+}
```

Comment:
Yes I also like defaults as contants to be honest. However I saw that no other default was stored as a constant so, to bring consistency, I decided to move the default for `retryCount` to the `GetEnvDefault()` call like with all the other defaults in the code, i.e. the field templates, format and TCP framing.

I have no problem to restore this. To make it a bit more consistent I can bring the defaults for `format` and `tcpFraming` also as package constants. The field templates defaults are trickier because the default for `tag` depends on `route.Options`.

> i think we can ignore the err from `strconv.Atoi`.

Sure! I don't like very much the excess of `err` checking needed in GoLang so I welcome this suggestion :)